example/vectorAdd/src/main.cpp

```diff
@@ -73,21 +73,42 @@ class VectorAddKernel
     }
 };
 
+template<class A, class B>
+#if defined(ALPAKA_ACC_CPU_BT_OMP4_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp4<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp2Blocks<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuFibers<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuOmp2Threads<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuSerial<A,B>;
+#elif defined(ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED)
+using DefaultAcc = alpaka::acc::AccCpuThreads<A,B>;
+#elif defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+using DefaultAcc = alpaka::acc::AccGpuCudaRt<A,B>;
+#else
+class Stub;
+#define NOP
+#warning "No supported backend selected."
+#endif
+
 auto main()
 -> int
 {
-// This example is hard-coded to use the sequential backend.
-#if defined(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+#ifndef NOP
```

Comment:
Currently the examples are hard-coded to use only one specific accelerator. This was done because my initial version which compiled the examples for all executors was seen as too complex and too verbose for beginners.
Now you want to use a middle way between "one hard-coded accelerator" and "all executors" which is "some default accelerator".
This will:
* make it a bit harder to understand the examples
* make it a bit easier to get the example to execute independent of your config

I would want to get this applied to all examples. The `DefaultAcc` selection code could be shared within a single header. One question would be where to store this header? Currently the examples are freestanding. It should be possible to copy them everywhere and they should still compile. However, such a `DefaultAcc` is nothing that should be part of the main `alpaka` library. It should probably also be renamed to `ExampleDefaultAcc` to not make it sound like something from the lib itself.

Reply:
I had the shared header in mind, when I wrote this. But I think it could be in alpaka to reduce the learning curve and similar backend selection code may be used helpful to reduce boilerplate in small alpaka application, which would have to replicate it otherwise.

Reply:
I do not think having a shared header necessarily prevents the examples from being freestanding. In case we put this header somewhere inside `alpaka/example/...`, the additional include path could be set in CMakeLists.txt for examples after `FIND_PACKAGE("alpaka" REQUIRED)`?

On whether such a header should be part of alpaka I lean towards @BenjaminW3 's view. However, in case there are lots of code like that, maybe worth considering making a new "alpakautil" library for such stuff.